aten/src/ATen/quantized/Quantizer.h

```diff
@@ -155,6 +155,62 @@ struct CAFFE2_API PerChannelAffineQuantizer : public AffineQuantizer {
   const int64_t axis_;
 };
 
+/**
+ * PerRowFloatQParamsQuantizer is the same as PerChannelAffineQuantizer
+ * except that it expects both scale and zero point to be floating point values.
+ *
+ * Note: Usage of floating point zero point is useful in cases where 0 doesn't need to
+ * be exactly represented in the quantized space. We can get additional precision by
+ * using floating point values for zero point.
+ */
+struct CAFFE2_API PerRowFloatQParamsQuantizer : public AffineQuantizer {
```

Comment:
since the fields and methods are the same, maybe this can inherit from PerChannelAffineQuantizer and reuse all the methods

Reply:
This will need a different qscheme and different quantize/dequantize functions. There is a comment in QScheme.h that each new qscheme should correspond 1-1 to a quantizer. https://github.com/pytorch/pytorch/blob/master/c10/core/QScheme.h

Reply:
yeah, we can still inherit from PerChannelAffineQuantizer for the implementation of this Quantizer to reduce code duplication and also make it more clear this qscheme is a variant of the per channel affine quantization.

Reply:
Dont we use the same quantizers for the symmetric and affine cases?

Reply:
no, we don't have symmetric quantized tensor in the backend. so we don't have symmetric quantizer.